src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl

```diff
@@ -216,12 +215,18 @@ send_to_index_queue(OrgId, Index, [SO|Rest], NameIdDict) ->
     PreliminaryEJson = decompress_and_decode(SO),
     NameKey = name_key(chef_object_type(Index)),
     ItemName = ej:get({NameKey}, PreliminaryEJson),
-    {ok, ObjectId} = dict:find(ItemName, NameIdDict),
-    StubRec = stub_record(Index, OrgId, ObjectId, ItemName, PreliminaryEJson),
-    %% Since we get here via valid `Index', we don't have to check that the objects are
-    %% indexable.
-    ok = oc_chef_object_db:add_to_solr_async(StubRec, PreliminaryEJson),
-    send_to_index_queue(OrgId, Index, Rest, NameIdDict).
+    NewMissingList = case dict:find(ItemName, NameIdDict) of
+        {ok, ObjectId} ->
+            StubRec = stub_record(Index, OrgId, ObjectId, ItemName, PreliminaryEJson),
+            %% Since we get here via valid `Index', we don't have to check that the objects are
+            %% indexable.
+            oc_chef_object_db:add_to_solr_async(StubRec, PreliminaryEJson),
+            [];
```

Comment:
This should be the unmodified `MissingList` getting returned - otherwise we empty our existing list